pkg/cache/cache_test.go

```diff
@@ -247,30 +240,23 @@ func testCacheEvictExpired(c ExpiringCache, t *testing.T) {
 }
 
 func testCacheEvicter(c ExpiringCache, t *testing.T) {
-	t.Skip("issue https://github.com/istio/istio/issues/4304")
 	c.SetWithExpiration("A", "A", 1*time.Millisecond)
 
-	// this is racy, but we're being generous enough that it should be fine
-	time.Sleep(10 * time.Millisecond)
-
-	_, ok := c.Get("A")
-	if ok {
-		t.Error("Got entry, expecting it to have been evicted")
-	}
-}
+	// loop until eviction happens. If eviction doesn't happen, this loop will get stuck forever which is fine
```

Comment:
This should use eventually...which I saw I still haven't merged. Meh. I'll get that PR in and then update this code.